src/driver_ros2.cpp

```diff
@@ -317,6 +317,24 @@ void DriverROS2::configureWrapper(const std::string & name)
   this->get_parameter_or("sync_mode", syncMode, std::string("standalone"));
   wrapper_->setSyncMode(syncMode);
   LOG_INFO("sync mode: " << syncMode);
+  bool trailFilter;
+  this->get_parameter_or("trail_filter", trailFilter, false);
+  std::string trailFilterType;
+  this->get_parameter_or("trail_filter_type", trailFilterType, std::string("trail"));
+  int trailFilterThreshold;
+  this->get_parameter_or("trail_filter_threshold", trailFilterThreshold, 0);
+  if (trailFilter) {
+    LOG_INFO("Using tail filter in " << trailFilterType << " mode with threshold " << trailFilterThreshold);
+    metavision_driver::MetavisionWrapper::TrailFilter::Type filter_type;
+    if (trailFilterType == "trail") {
+      filter_type = metavision_driver::MetavisionWrapper::TrailFilter::Type::TRAIL;
+    } else if (trailFilterType == "stc_trail_cut") {
+      filter_type = metavision_driver::MetavisionWrapper::TrailFilter::Type::STC_CUT_TRAIL;
+    } else if (trailFilterType == "stc_keep_cut") {
+      filter_type = metavision_driver::MetavisionWrapper::TrailFilter::Type::STC_KEEP_TRAIL;
+    }
```

Comment:
There is no final "else" clause, so filter_type will be left undefined.
I suggest removing all of this code, and passing trailFilterType straight through to the metavision wrapper

Reply:
But in this case, the ROS2 wrapper would need to be exposed to some metavision headers which it is not yet, afaik. This was the motivation between this "intermediate" type defined in the wrapper. Or is this related to a previous comment of using strings and mapping it at the end?